wsagent/che-core-api-languageserver-maven-plugin/src/main/java/org/eclipse/che/api/languageserver/generator/DtoGenerator.java

```diff
@@ -0,0 +1,300 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat - Initial Contribution
+ *******************************************************************************/
+package org.eclipse.che.api.languageserver.generator;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.reflections.Reflections;
+import org.reflections.scanners.SubTypesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+
+import com.google.common.base.Predicate;
+
+/**
+ * DtoGenerator generates che-style DTO's for lsp4j protocol classes.
+ * Subclasses specialize this class to generate client-side and server-side implementations.
+ * 
+ * @author Thomas Mäder
+ *
+ */
+public abstract class DtoGenerator {
```

Comment:
The typefox api's were using interfaces for protocol definition and used the general dto generator to generate dto classes.
LSP4j uses classes in the protocol defintion, and thus I had to write a new generator for toJson/fromJson conversion.
DtoGenerator is the base class for generating server side and client side dtos.